browser/ui/webui/playlist_ui.cc

```diff
@@ -25,6 +24,43 @@
 
 namespace playlist {
 
+namespace {
+
+class UntrustedPlayerUI : public ui::UntrustedWebUIController {
+ public:
+  explicit UntrustedPlayerUI(content::WebUI* web_ui)
+      : UntrustedWebUIController(web_ui) {
+    auto* source = CreateAndAddWebUIDataSource(
+        web_ui, kPlaylistPlayerURL, kPlaylistGenerated, kPlaylistGeneratedSize,
+        IDR_PLAYLIST_PLAYER_HTML);
+    source->AddFrameAncestor(GURL(kPlaylistURL));
+    source->OverrideContentSecurityPolicy(
+        network::mojom::CSPDirectiveName::ScriptSrc,
+        std::string("script-src 'self' chrome-untrusted://resources "
+                    "chrome-untrusted://brave-resources;"));
+    source->OverrideContentSecurityPolicy(
+        network::mojom::CSPDirectiveName::MediaSrc,
+        std::string("media-src 'self' chrome-untrusted://playlist-data "
+                    "https://*.googlevideo.com;"));
```

Comment:
why is this googlevideo.com? Couldn't a playlist contain videos from other sources?

Reply:
> Couldn't a playlist contain videos from other sources?

Yes, but we're focusing on Youtube at this moment. We'll have more sources in the future.

https://github.com/brave/brave-core/pull/15016#issuecomment-1250980893